lib/tree.rb

```diff
@@ -16,42 +16,136 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(n)
   def add(key, value)
```

Comment:
The space complexity is O(log n) if the tree is balanced and O(n) if it's not.
The method does work